src/common/snippets/src/generator.cpp

```diff
@@ -42,10 +44,21 @@ void Generator::generate(lowered::LinearIR& linear_ir, LoweringResult& result, c
     linear_ir.init_emitters(target);
 
     OV_ITT_TASK_NEXT(GENERATE, "::EmitCode")
-    auto loops2DKernel = std::make_shared<op::Kernel>(linear_ir);
-    loops2DKernel->compile_params = compile_params;
-    auto loops2DKernelExpr = linear_ir.create_expression(loops2DKernel, std::vector<lowered::PortConnectorPtr>{});
-    std::shared_ptr<Emitter> kernel = target->get(op::Kernel::get_type_info_static())(loops2DKernelExpr);
+
+    const auto contains_dynamic_exprs = std::any_of(linear_ir.cbegin(), linear_ir.cend(),
+                       [](const lowered::ExpressionPtr& expr) {
+                            return ov::is_type<op::LoopBeginDynamic>(expr->get_node()) ||
+                                   ov::is_type<op::LoopEndDynamic>(expr->get_node());
+                       });
+
+    std::shared_ptr<op::Kernel> kernel_op = nullptr;
+    if (linear_ir.is_dynamic() || contains_dynamic_exprs)
```

Comment:
As I understand we cannot make helper in `jit_kenrel_emitter.hpp` since it's CPU related code.
Anyway I agree that it'd be great if we have some builder for kernel. I added it to kernel node impl. Can you take a look please?

As for two conditions, I tried to make generic check (if we have internal dynamism). But seems like it's redundant at the moment. So let's leave only `contains_dynamic_exprs`